### ISSUE TITLE
Incorrect tilde version usage

### ISSUE THREAD

**Issue by pikeas** [python-poetry/poetry#2456]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.

- **OS version and name**: MacOS 10.15.3
- **Poetry version**: 1.0.5
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: See below

## Issue
Poetry fails to install `flask-appbuilder` in a Python 3.7 environment.

https://github.com/dpgaspar/Flask-AppBuilder/blob/6e77920a4dab5c62e8c37df8c343f136e75eff50/setup.py#L80

`python_requires="~=3.6",` is parsed by poetry as 3.6.x per https://github.com/python-poetry/poetry/issues/380, where the format actually means 3.x per https://packaging.python.org/guides/distributing-packages-using-setuptools/#semantic-versioning-preferred

Poetry's parsing of `~` in setup.py seems to be incorrect.

**Comment by finswimmer** [comment]
Hello @pikeas,

could you please provide your `pyproject.toml`?

fin swimmer

**Comment by finswimmer** [comment]
Hello,

you have a typo in `python = ">=3.7<3.8"`. It must be `python = ">=3.7,<3.8"`.

**Comment by pikeas** [comment]
@finswimmer Great catch, thank you!

Does Poetry not catch malformed version syntax?

**Comment by mkniewallner** [comment]
> Does Poetry not catch malformed version syntax?

It seems that this is the case now in to-be-released Poetry `1.2`.

Tested with [1.2.0b2](https://github.com/python-poetry/poetry/releases/tag/1.2.0b2):
```console
$ cat pyproject.toml | grep "python"
python = ">=3.7<3.8"

$ poetry install

Could not parse version constraint: >=3.7<3.8
```
